jamjar/jamjar/tasks/transcode_video.py

```diff
@@ -0,0 +1,60 @@
+
+from tasks import app
+
+import subprocess
+import logging
+import os
+import shutil
+
+import boto3
+
+# can't get this to work...
+# from django.conf import settings
+# if not settings.configured: settings.configure()
+
+def get_video_filepath(video_dir, extension, filename="video"):
+    full_filename = '{:}.{:}'.format(filename, extension)
+    return os.path.join(video_dir, full_filename)
+
+def transcode_to_hls(src, out):
+    logger = logging.getLogger(__name__)
+    result = subprocess.check_call(["ffmpeg", "-i", src, '-start_number', '0', '-hls_list_size', '0', '-f', 'hls', out])
+
+    if result == 0:
+        logger.info('Successfully transcoded {:} to {:}'.format(src, out))
+    else:
+        # this will retry the job
+        raise RuntimeError('Error transcoding {:} to {:}. Error code: {:}'.format(src, out, result))
+
+
+def upload_to_s3(src_dir):
+    s3 = boto3.resource('s3')
+
+    base_dir = os.path.basename(src_dir)
+    # TODO : make this use the actual environment 
+    s3_dir = os.path.join('prod', base_dir)
```

Comment:
`prod` should be taken from the settings file (so we can put them in different places for staging and whatnots)